packages/reactivity/src/effect.ts

```diff
@@ -396,16 +396,13 @@ export function triggerEffects(
 ) {
   // spread into array for stabilization
   const effects = isArray(dep) ? dep : [...dep]
-  for (const effect of effects) {
-    if (effect.computed) {
+  effects
+    .filter(effect => {
+      if (!effect.computed) return true
       triggerEffect(effect, debuggerEventExtraInfo)
-    }
-  }
-  for (const effect of effects) {
-    if (!effect.computed) {
-      triggerEffect(effect, debuggerEventExtraInfo)
-    }
-  }
+      return false
+    })
+    .forEach(effect => triggerEffect(effect, debuggerEventExtraInfo))
```

Comment:
This version seems to be much slower than the benchmarked version you provided earlier (and even slower than the current version in the codebase).

Reply:
You can compare it through the link below
[perf link](https://perf.link/%EF%BC%89%EF%BC%9AJSPerf#eyJpZCI6ImFvejkyNjFqa3kyIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGVmZmVjdHMgPSBbXVxuZm9yKGxldCBpPTA7aTwxMDA7aSsrKXtcbiAgZWZmZWN0cy5wdXNoKHtcbiAgIGNvbXB1dGVkOiBNYXRoLnJhbmRvbSgpPD0wLjVcbiAgfSlcbn1cbmNvbnN0IHRyaWdnZXJFZmZlY3QgPSAoKT0%2Be30iLCJ0ZXN0cyI6W3sibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJjb25zdCBub3RDb21wdXRlZEVmZmVjdHMgPSBlZmZlY3RzLmZpbHRlcih4ID0%2BIHtcbiAgaWYgKCF4LmNvbXB1dGVkKSByZXR1cm4gdHJ1ZVxuICB0cmlnZ2VyRWZmZWN0KHgpXG4gIHJldHVybiBmYWxzZVxufSlcblxuZm9yIChsZXQgaSA9IDA7IGkgPD0gbm90Q29tcHV0ZWRFZmZlY3RzLmxlbmd0aDsgKytpKSB7XG4gIHRyaWdnZXJFZmZlY3Qobm90Q29tcHV0ZWRFZmZlY3RzW2ldKVxufVxuIiwicnVucyI6WzI3MjAwMCwxMDAwLDI1NTAwMCwzMDgwMDAsNTAwMCwyMTgwMDAsMTY2MDAwLDMzNjAwMCw3ODAwMCwyMzcwMDAsNDgwMDAsMjE0MDAwLDg2MDAwLDY3MDAwLDEzMTAwMCwxNjkwMDAsNzUwMDAsMTUwMDAsNDEwMDAsMTUzMDAwLDYxMDAwLDQ2NjAwMCw1MTAwMCwxNDMwMDAsMjM2MDAwLDUwMDAsMjkzMDAwLDIxNTAwMCw5MzAwMCw5ODAwMCwxMDEwMDAsMTk4MDAwLDM5MDAwLDEwMDAsNjEwMDAsMjgwMDAsNjAwMCw3NzAwMCw4MTAwMCwyNTQwMDAsMTAyMDAwLDI3OTAwMCw0MzAwMCwxNjAwMCwxMjIwMDAsMzYxMDAwLDIwMDAwLDE5MDAwLDMwODAwMCwyNzgwMDAsMTg0MDAwLDI3OTAwMCw3NDAwMCwxMDAwLDc4MDAwLDEwMDAsNjkwMDAsMTAyMDAwLDc0MDAwLDE3NjAwMCw5MjAwMCwxNDUwMDAsMTA0MDAwLDcwMDAsMjYwMDAsMTYwMDAsNzkwMDAsMjU4MDAwLDIzMzAwMCw2OTAwMCwxNDAwMCwzMTEwMDAsMjkxMDAwLDM1MTAwMCwyOTAwMCwyMTAwMCw5NTAwMCwxMjcwMDAsNDAwMDAsMzcwMDAsMTc5MDAwLDQxMDAwLDE5MjAwMCw1MDAwLDEzNjAwMCwyMTIwMDAsMjYwMDAsMjgyMDAwLDgwMDAsMzQ1MDAwLDIzNzAwMCwxMDQwMDAsMzYzMDAwLDEzMDAwLDY2MDAwLDI1MzAwMCwxMDQwMDAsMjQ0MDAwLDI1NDAwMCw4MDAwMF0sIm9wcyI6MTM0NTcwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiY29uc3Qgbm90Q29tcHV0ZWRFZmZlY3RzID0gZWZmZWN0cy5maWx0ZXIoeCA9PiB7XG4gIGlmICgheC5jb21wdXRlZCkgcmV0dXJuIHRydWVcbiAgdHJpZ2dlckVmZmVjdCh4KVxufSlcblxuZm9yIChsZXQgaSA9IDA7IGkgPD0gbm90Q29tcHV0ZWRFZmZlY3RzLmxlbmd0aDsgKytpKSB7XG4gIHRyaWdnZXJFZmZlY3Qobm90Q29tcHV0ZWRFZmZlY3RzW2ldKVxufVxuIiwicnVucyI6WzE2MzAwMCwzMjgwMDAsMjgwMDAwLDI5MzAwMCw1MDAwLDgzMDAwLDE3OTAwMCwyOTAwMDAsMjA2MDAwLDEyNTAwMCw1ODAwMCwyNDMwMDAsNTcwMDAsNjIwMDAsNzIwMDAsMTYwMDAwLDcyMDAwLDEzMDAwLDY1MDAwLDMyNDAwMCw5MDAwLDIwMDAsMTE5MDAwLDQxMDAwMCwxMDgwMDAsNTAwMCwxODIwMDAsMTIxMDAwLDI0NTAwMCw1NDAwMCw3MTAwMCw4OTAwMCw1ODAwMCwxMDAwLDc0MDAwLDI1MDAwLDExMzAwMCw2MjAwMCw3MjAwMCwzMDcwMDAsMjE2MDAwLDkyMDAwLDM2MDAwLDE3MDAwLDE1MjAwMCw1NzAwMCwyMDAwMCwxOTAwMCwxMDYwMDAsMTY1MDAwLDgyMDAwLDExNjAwMCw2OTAwMCwyMTEwMDAsMTMxMDAwLDUwMDAsMjAyMDAwLDEwMTAwMCw1MjAwMCwyMzUwMDAsMjM5MDAwLDEwMTAwMCwyNjQwMDAsNzAwMCwyNjAwMCw0MDAwMCw1MDAwMCwxMDAwLDg2MDAwLDU4MDAwLDUwMDAsMjc3MDAwLDEzOTAwMCwzNDgwMDAsMzIwMDAsNDUwMDAsNzYwMDAsOTUwMDAsMjQzMDAwLDMzMDAwLDE0ODAwMCwzNjAwMCwxOTQwMDAsNjAwMCwxODAwMDAsMTM5MDAwLDE3MDAwLDQyNTAwMCwxODAwMCwzNzkwMDAsMTMwMDAwLDc3MDAwLDMyNjAwMCwxMzAwMCwzNzUwMDAsMTk3MDAwLDExMTAwMCwyMTQwMDAsMTAyMDAwLDEwNDAwMF0sIm9wcyI6MTI2NDUwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiZm9yKGNvbnN0IGVmZmVjdCBvZiBlZmZlY3RzLmZpbHRlcih4PT4ge1xuaWYoIXguY29tcHV0ZWQpIHJldHVybiB0cnVlXG4gIHRyaWdnZXJFZmZlY3QoeClcbiAgcmV0dXJuIGZhbHNlXG4gIH0pKSB7XG4gIHRyaWdnZXJFZmZlY3QoZWZmZWN0KVxuICB9IiwicnVucyI6WzIzMDAwLDI2NDAwMCwxMTYwMDAsMTYzMDAwLDUwMDAsMTg1MDAwLDEwNjAwMCwxNzIwMDAsMTExMDAwLDgwMDAwLDI3MDAwMCw5ODAwMCwyMzAwMCwzODAwMCw0MzAwMCwyNDQwMDAsMjc1MDAwLDE1OTAwMCw1ODAwMCwxODUwMDAsMzMwMDAsMTAwMCw0MTAwMCwxODgwMDAsNDkwMDAsNTAwMCwzODAwMCwxNTIwMDAsMjI3MDAwLDM0MDAwLDcyMDAwLDI1NzAwMCwzMzAwMCw5NDAwMCwyMjAwMCwyMDAwMCwxNzQwMDAsNDAwMDAsNzIwMDAsMTI3MDAwLDU2MDAwLDIxODAwMCwxODAwMCw1MDAwLDk1MDAwLDE1MzAwMCw4MDAwLDIwMDAsMTEzMDAwLDczMDAwLDcxMDAwLDYzMDAwLDU5MDAwLDMxMDAwLDkwMDAwLDI5MzAwMCwyNjAwMCw3NzAwMCw0MjAwMCwyNDQwMDAsMTYyMDAwLDUwMDAwLDgzMDAwLDYwMDAsMTkwMDAsNDYwMDAsMzQyMDAwLDE0NjAwMCw2MDAwMCw0MzAwMCw1MDAwLDE3MjAwMCwxNjMwMDAsMzE3MDAwLDM0MDAwLDU4MDAwLDg1MDAwLDc1MDAwLDE2MTAwMCwzMTAwMCw5MDAwMCwyMjAwMCwxMzgwMDAsNjAwMCw0NjAwMCw1NjAwMCwxODAwMCwxMDUwMDAsNTAwMCwxMTQwMDAsMTE3MDAwLDYxMDAwLDIxOTAwMCw2MDAwLDE5NjAwMCw1OTAwMCwyNDAwMCw0OTAwMCwxNTYwMDAsMTI4MDAwXSwib3BzIjo5NzA3MH0seyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImVmZmVjdHMuZmlsdGVyKHg9PiB7XG4gIGlmKCF4LmNvbXB1dGVkKSByZXR1cm4gdHJ1ZVxuICB0cmlnZ2VyRWZmZWN0KHgpXG4gIHJldHVybiBmYWxzZTtcbn0pLmZvckVhY2goeD0%2BIHRyaWdnZXJFZmZlY3QoeCkpIiwicnVucyI6WzMxMDAwLDE1NjAwMCwzMjMwMDAsMTU0MDAwLDUwMDAsNDQwMDAsMTQ2MDAwLDEzMzAwMCwzMjAwMCwxMTMwMDAsMzExMDAwLDQ0MDAwLDMyMjAwMCwyODAwMCw1NDAwMCwyOTgwMDAsMzkzMDAwLDIxNDAwMCw4MjAwMCwxNDMwMDAsMjIwMDAsMzM0MDAwLDI2NjAwMCwyMTAwMDAsNjAwMCw1MDAwLDUwMDAsNTAwMCwxNTQwMDAsMTEwMDAsMTAwMCwzMDcwMDAsMzA2MDAwLDM4MTAwMCwzMzAwMCw0MDcwMDAsMTYwMDAsNTYwMDAsMTY2MDAwLDcxMDAwLDI1ODAwMCw0MDAwMCw1MDAwLDEzMDAwLDE1NDAwMCwxODIwMDAsNTAwMCwyODUwMDAsODAwMCwxOTkwMDAsMjA4MDAwLDIwNDAwMCw2MTAwMCwyNTYwMDAsMzkwMDAsOTgwMDAsOTYwMDAsMTM3MDAwLDQxMzAwMCwxNjUwMDAsMTAwMDAsNTAwMCwzNDgwMDAsOTAwMCwzMjAwMCw1MDAwLDI1MDAwMCw0MzAwMCwxODMwMDAsNDkwMDAsMjAwMCw5NDAwMCwxODMwMDAsMTI1MDAwLDM4MDAwLDQ2MDAwLDQwMDAwLDM3MDAwLDE0MzAwMCwyMTAwMCwyMDYwMDAsMTAwMDAsNjYwMDAsNTAwMCw5NTAwMCwxOTUwMDAsNDAwMCw4MjAwMCw1MDAwLDI2NTAwMCwxOTMwMDAsNDEwMDAsMTEwMDAwLDUwMDAsMjgxMDAwLDI4MTAwMCwxODcwMDAsOTkwMDAsMTU4MDAwLDIxMjAwMF0sIm9wcyI6MTI3NzcwfSx7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6IiAgY29uc3Qgbm90Q29tcHV0ZWRFZmZlY3QgPSBbXVxuICBmb3IgKGNvbnN0IGVmZmVjdCBvZiBlZmZlY3RzKSB7XG4gICAgaWYgKGVmZmVjdC5jb21wdXRlZCkge1xuICAgICAgdHJpZ2dlckVmZmVjdChlZmZlY3QpXG4gICAgfSBlbHNlIHtcbiAgICAgIG5vdENvbXB1dGVkRWZmZWN0LnB1c2goZWZmZWN0KVxuICAgIH1cbiAgfVxuICBub3RDb21wdXRlZEVmZmVjdC5mb3JFYWNoKGVmZmVjdCA9PlxuICAgIHRyaWdnZXJFZmZlY3QoZWZmZWN0KVxuICApIiwicnVucyI6WzUwMDAsOTcwMDAsMTA4MDAwLDEwNzAwMCwyOTgwMDAsMTI1MDAwLDU1MDAwLDMxMDAwLDE1MDAwLDY1MDAwLDI1MDAwLDgwMDAsMTI4MDAwLDM0MTAwMCwyNTAwMCwzMTIwMDAsMjQwMDAwLDY5MDAwLDM5MDAwLDEwOTAwMCwyMzAwMCwxMzUwMDAsMTg5MDAwLDExMDAwLDE2NzAwMCwyNjAwMDAsMzgwMDAsMjgwMDAsNjAwMCw2MDAwLDE1MDAwLDg2MDAwLDQwMzAwMCwyNDEwMDAsNTAwMCw4NDAwMCw3MzAwMCwzMTAwMCw1NjAwMCwxMDUwMDAsNjQwMDAsOTIwMDAsMTM1MDAwLDUwMDAsMjAxMDAwLDM1MDAwLDIxMzAwMCw4NzAwMCwyNjYwMDAsMjExMDAwLDkxMDAwLDE4ODAwMCwyNTAwMCwxNjgwMDAsMTYyMDAwLDM1MzAwMCw4NDAwMCwxNTAwMCwyNjgwMDAsMTQzMDAwLDYzMDAwLDExMzAwMCw5NDAwMCwzMDgwMDAsMTEwMDAsNjAwMDAsMjUwMDAsNTAwMCwyNDAwMCwyNDAwMCw4MzAwMCw4NzAwMCwxMDAwLDc1MDAwLDIwMDAwLDI1MDAwLDEwMDAwLDE4NTAwMCw4MjAwMCwxNDAwMCw1MjAwMCwyNjgwMDAsMTUyMDAwLDEwMDAsNjUwMDAsNzQwMDAsMTkwMDAsMTUwMDAsMjI4MDAwLDExOTAwMCwxMDgwMDAsNzAwMDAsMjgzMDAwLDUwMDAsMTAyMDAwLDU2MDAwLDQwMDAwLDkzMDAwLDcwMDAwLDEzMDAwXSwib3BzIjoxMDExMjB9LHsibmFtZSI6IkZpbmQgaXRlbSAyMDAiLCJjb2RlIjoiZm9yIChjb25zdCBlZmZlY3Qgb2YgZWZmZWN0cykge1xuICAgIGlmIChlZmZlY3QuY29tcHV0ZWQpIHtcbiAgICAgIHRyaWdnZXJFZmZlY3QoZWZmZWN0KVxuICAgIH1cbiAgfVxuICBmb3IgKGNvbnN0IGVmZmVjdCBvZiBlZmZlY3RzKSB7XG4gICAgaWYgKCFlZmZlY3QuY29tcHV0ZWQpIHtcbiAgICAgIHRyaWdnZXJFZmZlY3QoZWZmZWN0KVxuICAgIH1cbiAgfVxuIiwicnVucyI6WzUwMDAsNDQwMDAsMTUyMDAwLDQwMDAsMTYwMDAsNzQwMDAsNTkwMDAsMTQ4MDAwLDE3MDAwMCwyMDcwMDAsMjExMDAwLDE3MDAwLDUwMDAsMTMwMDAsMjQwMDAsMTgwMDAsMTgxMDAwLDE2MDAwMCwyOTAwMCw3MDAwMCwxNTAwMCw1MDAwLDE0NDAwMCw0MDAwMCw4ODAwMCwxMTAwMCwzMDAwMCw1MDAwLDY3MDAwLDEwMDAsNTkwMDAsNTAwMDAsMTI2MDAwLDExODAwMCw1MDAwLDUwMDAsMTM2MDAwLDI2MDAwLDQzMDAwLDc2MDAwLDI1MDAwLDQzMDAwLDczMDAwLDExMDAwLDEzMjAwMCw5MDAwLDE2NzAwMCwxMDAwLDE0MjAwMCwxMzAwMCw1MjAwMCwxNjAwMCwxMzAwMCwxMzQwMDAsMTEwMDAsNzUwMDAsNDkwMDAsMzAwMDAsMTQyMDAwLDUwMDAsMjEwMDAsNDYwMDAsODYwMDAsMTI4MDAwLDEzMDAwMCwxOTkwMDAsMTE2MDAwLDEyNjAwMCw2NDAwMCwxNjAwMCw3MDAwLDIwMzAwMCwzNzAwMCwxMDkwMDAsNTAwMCwxODMwMDAsMTYyMDAwLDU5MDAwLDUxMDAwLDkwMDAsMTYwMDAsNDcwMDAsMzAwMCwxMjcwMDAsMTg5MDAwLDU5MDAwLDYyMDAwLDE4MDAwLDE5MDAwLDgxMDAwLDcwMDAwLDQ0MDAwLDE2MDAwLDIxNTAwMCw5NzAwMCw2MDAwLDE3MDAwLDE2MDAwLDEzNDAwMCw0NDAwMF0sIm9wcyI6NjgzNzB9XSwidXBkYXRlZCI6IjIwMjMtMTEtMTNUMDc6NDI6MTcuODgyWiJ9)